src/_bentoml_sdk/service/factory.py

```diff
@@ -276,8 +276,14 @@ def gradio_app_startup_hook(self, max_concurrency: int):
         if gradio_apps:
             for gradio_app, path, name in gradio_apps:
                 logger.info(f"Initializing gradio app at: {path or '/'}")
-                gradio_app.get_blocks().queue(default_concurrency_limit=max_concurrency)
-                gradio_app.get_blocks().startup_events()
+                blocks = gradio_app.get_blocks()
+                blocks.queue(default_concurrency_limit=max_concurrency)
+                if hasattr(blocks, "startup_events"):
```

Comment:
should we support gradio 5.0 only?

for folks who need to write a new BentoML service they prob should also upgrade their gradio interface to new version?

ig for ppl who alr have gradio 4.0 then 🤔